docs/implement-a-customized-actuator-en.md

```diff
@@ -0,0 +1,322 @@
+
+The logic for `SumContract` is the summation of two numerical values:
+
+```protobuf
+syntax = "proto3";
+package protocol;
+option java_package = "org.tron.protos.contract"; //Specify the name of the package that generated the Java file
+option go_package = "github.com/tronprotocol/grpc-gateway/core";
+message SumContract {
+    int64 param1 = 1;
+    int64 param2 = 2;
+    bytes owner_address = 3;
+}
+```
+
+Meanwhile, register the new contract type in `Transaction.Contract.ContractType` emuneration within the src/main/protos/core/Tron.proto file. Important data structures, such as transactions, accounts and blocks are defined in the Tron.proto file:
+
+```protobuf
+message Transaction {
+  message Contract {
+    enum ContractType {
+      AccountCreateContract = 0;
+      TransferContract = 1;
+      ........
+      SumContract = 52;  
+    }
+  ...
+}
+```
+
+Then register a function to ensure that gRPC can receive and identify the requests of this contracts. Currently, gRPC protocols are all defined in src/main/protos/api/api.proto. Add an `InvokeSum` interface in Wallet Service within api.proto:
+
+```protobuf
+service Wallet {
+  rpc InvokeSum (SumContract) returns (Transaction) {
+    option (google.api.http) = {
+      post: "/wallet/invokesum"
+      body: "*"
+      additional_bindings {
+        get: "/wallet/invokesum"
+      }
+    };
+  };
+  ...
+};
+```
+At last, recompile the modified proto files. directly compiling the java-tron project to compile the proto files is recommended, `protoc` command is also supported.
+
+*Currently, java-tron uses protoc v3.4.0. Please keep the same version when compiling by `protoc` command.*
+
+```shell
+# recommended
+./gradlew build -x test
+
+# or build via protoc
+protoc -I=src/main/protos -I=src/main/protos/core --java_out=src/main/java  Tron.proto
+protoc -I=src/main/protos/core/contract --java_out=src/main/java  math_contract.proto
+protoc -I=src/main/protos/api -I=src/main/protos/core -I=src/main/protos  --java_out=src/main/java api.proto
+```
+
+After compilation, the corresponding .class under the java_out directory will be updated.
+
+## Implement SumActuator
+
+For now, the default Actuator supported by java-tron is located in `org.tron.core.actuator` of this module. Creating `SumActuator` under this directory:
+
+```java
+public class SumActuator extends AbstractActuator {
+
+  public SumActuator() {
+    super(ContractType.SumContract, SumContract.class);
+  }
+
+  /**
+   * define the contract logic in this method
+   * e.g.: do some calculate / transfer asset / trigger a contract / or something else
+   *
+   * SumActuator just sum(param1+param2) and put the result into logs.
+   * also a new chainbase could be created to store the generated data(how to create a chainbase will be released in future.)
```

Comment:
s/could/can/

Reply:
how to create a chainbase will be released in future =>

released => revealed

since it's not a release.

Or you can say:

Document about how to xxxx will be released in the near future